tilings/strategies/requirement_placement.py

```diff
@@ -453,6 +459,10 @@ def req_indices_and_directions_to_place(
         else:
             all_reqs = tiling.requirements
         for req in all_reqs:
+            if self.max_rules_per_req is not None:
+                num_rule = reduce(lambda prod, req: prod * len(req), req, 1)
```

Comment:
should multiply by 4 for directions

Reply:
I thought about that, it depends on how we want this flag to play with the `dirs` flag. Maybe we should multiply by the number of directions.

Reply:
Good point, it should multiple len(dirs) as you suggested

Reply:
Fixed.